docs/literate.jl

```diff
@@ -0,0 +1,56 @@
+# Retrieve filename of literate script
+if length(ARGS) != 2
+    error("please specify the literate script and the output directory")
+end
+const SCRIPTJL = ARGS[1]
+const OUTDIR = ARGS[2]
+
+# Activate environment
+using Pkg: Pkg
+Pkg.activate(dirname(SCRIPTJL))
```

Comment:
Requiring the examples scripts to include Literate in their dependencies when they don't actually use it seems weird; maybe that's the best we can do (and of course we can then clarify that in the examples/README.md), but why not add them here as well:
```suggestion
Pkg.activate(dirname(SCRIPTJL))
Pkg.add("Literate")
```
this should add it to the correct Project.toml (e.g. when I write a new examples script and locally run make.jl), but be a no-op if it's already there.

Reply:
I thought about this as well and I am still not sure what's the best approach. The advantage of adding them to the Project.toml + Manifest.toml is that it is correctly tracked and updated by CompatHelper and that it is guaranteed that the installation of Literate does not downgrade any other packages or mess with the package setup which would impact reproducibility if it is not part of Project.toml + Manifest.toml (otherwise it might only be installed when the CI tests are run).

Reply:
Would
```suggestion
Pkg.activate(dirname(SCRIPTJL))
Pkg.add("Literate"; preserve=PRESERVE_ALL)
```
or similar ([Pkg.add docs](https://pkgdocs.julialang.org/v1/api/#Pkg.add)) not satisfy that?